apps/hash-external-services/temporal/setup.sh

```diff
@@ -106,30 +61,32 @@ setup_server() {
   if [[ ${SKIP_DEFAULT_NAMESPACE_CREATION} != true ]]; then
     register_default_namespace
   fi
-
-  if [[ ${SKIP_ADD_CUSTOM_SEARCH_ATTRIBUTES} != true ]]; then
-    add_custom_search_attributes
-  fi
 }
 
 # === Main ===
 
-# TEMPORAL_CLI_ADDRESS setup copied from https://github.com/temporalio/docker-builds/blob/main/docker/entrypoint.sh
 
-: "${BIND_ON_IP:=$(getent hosts "$(hostname)" | awk '{print $1;}')}"
+# TEMPORAL_ADDRESS setup copied from https://github.com/temporalio/docker-builds/blob/main/docker/entrypoint.sh
+
+: "${BIND_ON_IP:=$(getent hosts $(hostname) | awk '{print $1;}')}"
 export BIND_ON_IP
 
-# check TEMPORAL_CLI_ADDRESS is not empty
-if [[ -z "${TEMPORAL_CLI_ADDRESS:-}" ]]; then
-    echo "TEMPORAL_CLI_ADDRESS is not set, setting it to ${BIND_ON_IP}:7233"
+# check TEMPORAL_ADDRESS is not empty
+if [[ -z "${TEMPORAL_ADDRESS:-}" ]]; then
+    echo "TEMPORAL_ADDRESS is not set, setting it to ${BIND_ON_IP}:7233"
 
     if [[ "${BIND_ON_IP}" =~ ":" ]]; then
         # ipv6
-        export TEMPORAL_CLI_ADDRESS="[${BIND_ON_IP}]:7233"
+        export TEMPORAL_ADDRESS="[${BIND_ON_IP}]:7233"
     else
         # ipv4
-        export TEMPORAL_CLI_ADDRESS="${BIND_ON_IP}:7233"
+        export TEMPORAL_ADDRESS="${BIND_ON_IP}:7233"
     fi
 fi
 
+# TEMPORAL_CLI_ADDRESS is deprecated and support for it will be removed in the future release.
+if [[ -z "${TEMPORAL_CLI_ADDRESS:-}" ]]; then
+    export TEMPORAL_CLI_ADDRESS="${TEMPORAL_ADDRESS}"
+fi
```

Comment:
As we use `TEMPORAL_ADDRESS` above why we need to set the deprecated environment variable?

Reply:
We don't really need it. It does allow us to downgrade temporal in case we need to use a previous version, though. It was useful for me when I tested the migration path from 1.20 to 1.21 to ensure that our system was still oeprational. The official scripts also do this and I thought it would be good to follow their example.